lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -118,12 +118,22 @@ def add_address(post, card, options)
         address_object = {}
         address_object[:state] = address[:state] if address[:state]
         address_object[:city] = address[:city] if address[:city]
-        address_object[:zip_code] = address[:zip_code] if address[:zip_code]
-        address_object[:street] = address[:street] if address[:street]
-        address_object[:number] = address[:number] if address[:number]
+        address_object[:zip_code] = address[:zip] if address[:zip]
+        address_object[:street] = parse_street(address) if address[:address1]
+        address_object[:number] = parse_house_number(address) if address[:address1]
```

Comment:
do we care about backwards compatibility? or since the existing code is very non-standard compared to others, do we feel fine removing support for those old fields

Reply:
👍 I have added `address[:street]` and `address[:number]` back in for backwards compatibility/ allowing direct assignment.